utils/build.ps1

```diff
@@ -761,8 +820,12 @@ function Invoke-IsolatingEnvVars([scriptblock]$Block) {
   }
 }
 
-function Invoke-VsDevShell($Arch) {
-  $DevCmdArguments = "-no_logo -host_arch=$($BuildArch.VSName) -arch=$($Arch.VSName)"
+function Invoke-VsDevShell([Hashtable] $Platform) {
+  if (($Platform.OS -ne [OS]::Windows) -or ($BuildPlatform.OS -ne [OS]::Windows)) {
+    throw "Invoke-VsDevShell is only available on Windows."
```

Comment:
Should this throw? If we simply report a warning and return, it might just work on other hosts.

Reply:
No, I mean't make this function a no-op if the build is not for a toolchain to be hosted on Windows.